lesson15/main.js

```diff
@@ -0,0 +1,214 @@
+const numberLabel = document.getElementById("js-number-label");
+const nameBox = document.getElementById("js-name-box");
+const numberBox = document.getElementById("js-number-box");
+const namePattern =
+  /^[ぁ-んァ-ヶｱ-ﾝﾞﾟ一-龠a-zA-Zａ-ｚＡ-Ｚ]+[ぁ-んァ-ヶｱ-ﾝﾞﾟ一-龠a-zA-Zａ-ｚＡ-Ｚ\s]*$/;
+//https://arc-tech.hatenablog.com/entry/2021/01/20/105620
+const numberPattern = /^-?\d+(\.?\d*)([eE][+-]?\d+)?$/;
+const url = "https://mocki.io/v1/1c058349-634e-462a-ad37-14f135e59b99";
+
+function renderStatus(response) {
+  errorMessage.id = "render-status";
+  errorMessage.textContent = `${response.status}:${response.statusText}`;
+  document.body.appendChild(errorMessage);
+}
+
+function displayInfo(error) {
+  errorMessage.id = "display-info";
+  errorMessage.textContent = error;
+  document.body.appendChild(errorMessage);
+}
+
+function renderCircle() {
+  const loadingCircle = document.createElement("img");
+  loadingCircle.src = "img/loading-circle.gif";
+  loadingCircle.alt = "ローディング画像";
+  loadingCircle.id = "loading-circle";
+  ul.appendChild(loadingCircle);
+}
+
+function removeCircle() {
+  document.getElementById("loading-circle").remove();
+}
+
+function renderData(menus) {
+  const fragment = document.createDocumentFragment();
+  fragment.id = "fragment";
+  for (const menu of menus) {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+    a.href = menu.to;
+    a.textContent = menu.text;
+    img.src = menu.img;
+    img.alt = menu.alt;
+    fragment
+      .appendChild(li)
+      .appendChild(a)
+      .insertAdjacentElement("afterbegin", img);
+  }
+  ul.appendChild(fragment);
+}
+
+async function fetchData(url) {
+  renderCircle();
+  try {
+    const response = await fetch(url);
+    const responseData = await response.json();
+    if (!response.ok) {
+      renderStatus(response);
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    if (!responseData.length) {
+      displayInfo("no data");
+    }
+    return responseData;
+  } catch (error) {
+    displayInfo(error);
+  } finally {
+    removeCircle();
+    backButton.classList.remove("hidden");
+  }
+}
+
+async function fetchRenderData(inputName, inputNumber) {
+  const responseData = await fetchData(url);
+  if (responseData) {
+    renderData(responseData);
+    console.log(inputName, inputNumber); //show inputted number
+  }
+}
+
+openButton.addEventListener("click", () => {
+  promptMessage.textContent = "入力後、取得ボタンを押してね";
+  promptMessage.style.color = "black";
+  nameBox.value = "";
+  numberBox.value = "";
+  modal.classList.remove("hidden");
+  mask.classList.remove("hidden");
+  openButton.classList.add("hidden");
+  fetchButton.setAttribute("disabled", "true");
+  nameLabel.textContent = "名前";
+  nameLabel.style.color = "black";
+  numberLabel.textContent = "数字";
+  numberLabel.style.color = "black";
+  nameBox.focus();
+});

```

Comment:
IMO
I thought it was unnecessary.

```suggestion
openButton.addEventListener("click", () => {
  nameBox.value = "";
  numberBox.value = "";
  modal.classList.remove("hidden");
  mask.classList.remove("hidden");
  openButton.classList.add("hidden");
  fetchButton.setAttribute("disabled", "true");
  nameBox.focus();
});

```

Reply:
thank you haru-san :)

It’s neccesasry .

Reply:
thank you haru-san :)
I coded that for resetting prompt message.
[rec-area (1).webm]

Reply:
@groovywave 
in which case, You can use existing functions :)

```
openButton.addEventListener("click", () => {
  resetPrompt(); // here
  nameBox.value = "";
  numberBox.value = "";
  modal.classList.remove("hidden");
  mask.classList.remove("hidden");
  openButton.classList.add("hidden");
  fetchButton.setAttribute("disabled", "true");
  nameBox.focus();
});
```

Reply:
> @groovywave in which case, You can use existing functions :)
> 
> ```
> openButton.addEventListener("click", () => {
>   resetPrompt(); // here
>   nameBox.value = "";
>   numberBox.value = "";
>   modal.classList.remove("hidden");
>   mask.classList.remove("hidden");
>   openButton.classList.add("hidden");
>   fetchButton.setAttribute("disabled", "true");
>   nameBox.focus();
> });
> ```

Thank you haru-san :)
I fixed that.
I wrote resetPrompt() .
I couldn't commit properly ;<